src/main/java/com/github/mmolimar/kafka/connect/fs/policy/AbstractPolicy.java

```diff
@@ -105,16 +109,25 @@ private String convert(String uri) {
         if (hasEnded()) {
             throw new IllegalWorkerStateException("Policy has ended. Cannot be retried.");
         }
+        
+        if (batchSize > 0 && currentIterator != null && currentIterator.hasNext()) {
```

Comment:
I think it's better to avoid this check and set the ``files`` iterator with the ``previous`` value

Reply:
If we skip this check, the `preCheck` and `postCheck` methods will be executed which will cause the `CronPolicy`, `SleepyPolicy` and `HdfsFileWatcherPolicy` to sleep between batches which breaks the current behaviour of sleeping only after handling all the files. Is this what you want?

Reply:
Makes totally sense but just returning ``previous`` (in case it has elements and ignoring the batch size), ok?

Reply:
That sounds ok. 
Since `previous` is the original iterator (not `BatchIterator`) I will need to create a custom class extending Iterator with a method to reset the counter to `0`. Then we can store the `BatchIterator` in the `previous` variable instead. 

I will change it tomorrow.

Reply:
Changed it to use `com.google.common.collect.Iterators.partition` instead which simplified the implementation and becomes more standard.